File: src/main/java/com/github/zoewithabang/model/MessageData.java
package com.github.zoewithabang.model;

import java.sql.Timestamp;
import java.util.Objects;

public class MessageData implements IModel
{
    private String id;
    private String userId;
    private String content;
    private Long timestamp;
    
    public MessageData(){}
    
    public MessageData(String id, String userId, String content, Long timestamp)
    {
        this.id = id;
        this.userId = userId;
        this.content = content;
        this.timestamp = timestamp;
    }
    
    public MessageData(String id, String userId, String content, Timestamp timestamp)
    {
        this.id = id;
        this.userId = userId;
        this.content = content;
        this.timestamp = timestamp.getTime();
    }
    
    public String getId()
    {
        return id;
    }
    
    public void setId(String id)
    {
        this.id = id;
    }
    
    public String getUserId()
    {
        return userId;
    }
    
    public void setUserId(String userId)
    {
        this.userId = userId;
    }
    
    public String getContent()
    {
        return content;
    }
    
    public void setContent(String content)
    {
        this.content = content;
    }
    
    public Long getTimestampLong()
    {
        return timestamp;
    }
    
    public Timestamp getTimestampTimestamp()
    {
        return new Timestamp(timestamp);
    }
    
    public void setTimestamp(Long timestamp)
    {
        this.timestamp = timestamp;
    }
    
    public void setTimestamp(Timestamp timestamp)
    {
        this.timestamp = timestamp.getTime();
    }
    
    @Override
    public String toString()
    {
        return "MessageData{" +
            "id='" + id + '\'' +
            ", userId='" + userId + '\'' +
            ", content='" + content + '\'' +
            ", timestamp=" + timestamp +
            '}';
    }
    
    @Override
    public boolean equals(Object o)
    {
        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;
        MessageData that = (MessageData) o;
        return Objects.equals(id, that.id) &&
            Objects.equals(userId, that.userId) &&
            Objects.equals(content, that.content) &&
            Objects.equals(timestamp, that.timestamp);
    }
    
    @Override
    public int hashCode()
    {
        return Objects.hash(id, userId, content, timestamp);
    }
}

File: src/main/kotlin/com/github/zoewithabang/model/MessageData.kt
package com.github.zoewithabang.model

import java.sql.Timestamp

class MessageData {
    var id: String? = null
    var userId: String? = null
    var content: String? = null
    var timestampLong: Long? = null
        private set

    val timestampTimestamp: Timestamp?
        get() = timestampLong?.let { Timestamp(it) }

    constructor(id: String, userId: String, content: String, timestamp: Long) {
        this.id = id
        this.userId = userId
        this.content = content
        this.timestampLong = timestamp
    }

    constructor(id: String, userId: String, content: String, timestamp: Timestamp) {
        this.id = id
        this.userId = userId
        this.content = content
        this.timestampLong = timestamp.time
    }

    override fun toString(): String {
        return "MessageData(id=$id, userId=$userId, content=$content, timestampLong=$timestampLong)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MessageData

        if (id != other.id) return false
        if (userId != other.userId) return false
        if (content != other.content) return false
        if (timestampLong != other.timestampLong) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (userId?.hashCode() ?: 0)
        result = 31 * result + (content?.hashCode() ?: 0)
        result = 31 * result + (timestampLong?.hashCode() ?: 0)
        return result
    }
}
